### PR TITLE
Jenkins build script for Linux

### PR DESCRIPTION
**Pull request by jaredpar**
This PR adds cibuild.sh to the repo which is used to run our Linux Jenkins job.  The script is very simple at this point:

- Update NuGet packages.
- Build csc using mono + xbuild.

Other scenarios are going to be coming online over the next few weeks.  Many of which we already know to work but need to nail down the test infrastructure for.

### CODE REVIEW

File: cibuild.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+echo Changing mono snapshot
+. mono-snapshot mono/20150316155603
+if [ $? -ne 0 ]; then
```
Comment (jasonmalinowski) [inline cibuild.sh]: Congrats for being the first bash script in history to properly think about error handling.

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by jaredpar** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@jaredpar I'm seeing slave timeouts after the upgrade to the latest Jenkins. Investigating

**Comment by jaredpar** [comment]
@mmitche one of the slave machines looked like it got into that bad file system state.  I yanked it offline to see if it would unstick the queue.

**Comment by mmitche** [comment]
@jaredpar this is because the job aborted because the machine went "offline" but continued to run the job.

**Pull request merged**